File: Vishwanathan-Sabareesh-HW2/shapes/src/Rectangle.java
import java.util.List;

public class Rectangle extends Quadrilateral implements SymmetricTwoDShape {
    private final TwoDPoint[] vertices = new TwoDPoint[4];

    public Rectangle(double... vertices) {
        this(TwoDPoint.ofDoubles(vertices));
    }

    public Rectangle(List<TwoDPoint> vertices){ super(vertices); }

    /**
     * The center of a rectangle is calculated to be the point of intersection of its diagonals.
     *
     * @return the center of this rectangle.
     */
    @Override
    public Point center() {
        TwoDPoint center = new TwoDPoint((vertices[0].getX() + vertices[2].getX())/2, (vertices[0].getY() + vertices[2].getY())/2);
        return null; // TODO
    }

    @Override
    public boolean isMember(List<? extends Point> vertices) {
        double x = ((TwoDPoint)vertices.get(0)).getX();
        double y = ((TwoDPoint)vertices.get(1)).getX();
        if(!(((TwoDPoint)vertices.get(0)).getX() == ((TwoDPoint)vertices.get(3)).getX() &&
                ((TwoDPoint)vertices.get(1)).getX() == ((TwoDPoint)vertices.get(2)).getX()))
            return false;
        else if(!(Math.abs(((TwoDPoint)vertices.get(1)).getX() - ((TwoDPoint)vertices.get(0)).getX()) ==
                Math.abs(((TwoDPoint)vertices.get(3)).getX() - ((TwoDPoint)vertices.get(2)).getX()) &&
                Math.abs(((TwoDPoint)vertices.get(1)).getY() - ((TwoDPoint)vertices.get(2)).getY()) ==
                        Math.abs(((TwoDPoint)vertices.get(0)).getY() - ((TwoDPoint)vertices.get(3)).getY())))
            return false;
        return true;// TODO
    }

    @Override
    public double area() {
        double[] sideLengths = getSideLengths();
        return sideLengths[0]*sideLengths[1]; // TODO
    }
}

File: Vishwanathan-Sabareesh-HW3/rectangle.py
from quadrilateral import Quadrilateral
from two_d_point import TwoDPoint


class Rectangle(Quadrilateral):

    def __init__(self, *floats):
        super().__init__(*floats)
        if not self.__is_member():
            raise TypeError("A rectangle cannot be formed by the given coordinates.")

    def __eq__(self, other):
        if isinstance(other, self.__class__):
            return self.vertices == other.vertices
        return False

    def __ne__(self, other):
        return not self.__eq__(other)

    def __str__(self):
        return '(%g, %g), (%g, %g), (%g, %g), (%g, %g)' % (
            self.vertices[0].x, self.vertices[0].y, self.vertices[1].x, self.vertices[1].y,
            self.vertices[2].x, self.vertices[2].y, self.vertices[3].x, self.vertices[3].y)

    def center(self):
        """Returns the center of this rectangle, calculated to be the point of intersection of its diagonals."""
        center = TwoDPoint((self.vertices[0].x + self.vertices[2].x)/2, (self.vertices[0].y + self.vertices[2].y)/2)
        return center# TODO

    def area(self):
        """Returns the area of this rectangle. The implementation invokes the side_lengths() method from the superclass,
        and computes the product of this rectangle's length and width."""
        side_lengths = self.side_lengths()
        return side_lengths[0]*side_lengths[1]  # TODO

    def __is_member(self):
        if len(self.vertices) != 4:
            return False
        if self.vertices[0].y != self.vertices[1].y or self.vertices[2].y != self.vertices[3].y or \
                self.vertices[0].x != self.vertices[3].x or self.vertices[1].x != self.vertices[2].x:
            return False
        return True  # TODO